src/devcenter/azext_devcenter/manual/custom.py

```diff
@@ -1134,16 +1135,25 @@ def devcenter_dev_box_delay_upcoming_action(
     dev_center,
     project_name,
     dev_box_name,
+    delay_time,
     upcoming_action_id,
-    delay_until,
     user_id='me'
 ):
+    split_time = delay_time.split(':')
+    hours = int(split_time[0])
+    minutes = int(split_time[1])
     cf_dataplane = cf_devcenter_dataplane(cmd.cli_ctx, dev_center, project_name)
+    upcoming_action = cf_dataplane.dev_box.get_upcoming_action(
+        user_id=user_id,
+        dev_box_name=dev_box_name,
+        upcoming_action_id=upcoming_action_id,
+    )
+    delayed_time = upcoming_action.scheduled_time + timedelta(hours=hours, minutes=minutes)
```

Comment:
Just to confirm, so we allow the user to delay from the new scheduled time and not the original?

Reply:
Yes!